### ISSUE TITLE
Getting redirected to "/js/routing?callback=fos.Router.setData" after logging in

### ISSUE THREAD

**Issue by ls-adrian** [FriendsOfSymfony/FOSJsRoutingBundle#140]
I'm redirecting to the `target_path` in my `success_handler` but in this case it's the above mentioned URL. This is an issue because I'd rather not be redirecting the user to a JS file.

Anyone have any ideas on how to resolve this?

**Comment by inmarelibero** [comment]
@ls-adrian how did you do exactly?

**Comment by ls-adrian** [comment]
@inmarelibero I added this to my `security.yml` file

```yml
    firewalls:
        js_router:
            pattern: ^/(js\/routing)
            security: false
```

**Comment by ToshY** [comment]
Why is this closed exactly? I'm using Symfony 5.2.10 and try to exclude like @ls-adrian suggested, but that doesn't seem to have  any impact. The only workaround I've found that works, is that I have to add  `... || request->getPathInfo() === '/js/routing'` in the `onKernelRequest` in my `RequestSubscriber`.

**Comment by DevShaun1** [comment]
Still doesn't appear to be an official fix for this issue. My solution was to create an event listener class
`// src/EventListener/JsRoutingAccessListener.php

namespace App\EventListener;

use Symfony\Component\HttpKernel\Event\RequestEvent;
use Symfony\Component\Security\Core\Security;
use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;

class JsRoutingAccessListener
{
    private $security;

    public function __construct(Security $security)
    {
        $this->security = $security;
    }

    public function onKernelRequest(RequestEvent $event)
    {
        $request = $event->getRequest();

        // Check if the route matches the JS routing route
        if ($request->attributes->get('_route') === 'fos_js_routing_js') {
            // Restrict access if the user is not logged in
            if (!$this->security->isGranted('ROLE_USER')) {
                throw new AccessDeniedHttpException('You must be logged in to access this resource.');
            }
        }
    }
}`

Then register the listener in the service.yaml file:
`services:
    App\EventListener\JsRoutingAccessListener:
        tags:
            - { name: kernel.event_listener, event: kernel.request, priority: 0 }`
